src/cheri/app-cap-description.adoc

```diff
@@ -591,6 +591,10 @@ NOTE: These checks are much less rigorous than parity or ECC protection, and are
 NOTE: Even though valid capabilities which fail the integrity check could not have been legally generated,
  defining the handling in the architecture allows the behavior to be precisely specified for all 2^(CLEN+1)^ input values.
 
+Implementing these checks is optional, as all detected cases are theoretically unreachable.
+
+IMPORTANT: The default is to _include_ integrity checks, and they are _excluded_ in all cases to save logic area if {cheri_nointegrity_ext} is implemented.
```

Comment:
I feel like we shouldn't introduce a new extension. Maybe something like this?
```suggestion
IMPORTANT: It is strongly recommended to _include_ integrity checks, but they may be omitted to save area in systems that do not wish to be more resilient to bit flips.
```

Reply:
And maybe some note on needing to define behaviour for all combination of input bits for DV?

Reply:
I don't think resistance to bit flips is what we're aiming for here - if you want that you put ECC everywhere.
I think what we're aiming for is capabilities written by an incompatible IP

Reply:
If it’s possible to do so then that is an integration error that does not conform to the RISC-V specification.

Reply:
I'm not sure we want to deny such systems, I've heard people talking of (e.g.) mixed 32-bit/64-bit CHERI systems before.
I'm advocating for such a system, but I don't see why we should prevent it by definition.

So these checks will be saying that a capability has been read which can't be handled, which is probably an error in a heterogenous system, but better to have a definition of the behaviour in these cases.

Reply:
You can totally do that if you design your system very carefully, but the tags must be distinct. What you cannot do is interpret a valid 64-bit capability as a 128-bit one, or vice-versa, and get a valid tag.

Reply:
I think the current wording reflects that as it talks about `IP compatibility issues` so any objections to merging?

Reply:
New wording is vague enough. Sounds good to me